jscomp/build_tests/cli_help/input.js

```diff
@@ -0,0 +1,202 @@
+  "\n" +
+  "Options:\n" +
+  "  -w          Watch mode\n" +
+  "  -ws         [host]:port set up host & port for WebSocket build notifications\n" +
+  "  -verbose    Set the output to be verbose\n" +
+  "  -with-deps  *deprecated* This is the default behavior now. This option will be removed in a future release\n";
+
+const cleanHelp =
+  "Usage: rescript clean <options>\n" +
+  "\n" +
+  "`rescript clean` cleans build artifacts\n" +
+  "\n" +
+  "Options:\n" +
+  "  -verbose    Set the output to be verbose\n" +
+  "  -with-deps  *deprecated* This is the default behavior now. This option will be removed in a future release\n";
+
+const formatHelp =
+  "Usage: rescript format <options> [files]\n" +
+  "\n" +
+  "`rescript format` formats the current directory\n" +
+  "\n" +
+  "Options:\n" +
+  "  -stdin  [.res|.resi|.ml|.mli] Read the code from stdin and print\n" +
+  "          the formatted code to stdout in ReScript syntax\n" +
+  "  -all    Format the whole project \n" +
+  "  -check  Check formatting for file or the whole project. Use `-all` to check the whole project\n";
+
+const convertHelp =
+  "Usage: rescript convert <options> [files]\n" +
+  "\n" +
+  "`rescript convert` converts the current directory\n" +
+  "\n" +
+  "**This command removes old OCaml files and creates new ReScript \n" +
+  "files. Make sure your work is saved first!**\n" +
+  "\n" +
+  "Options:\n" +
+  "  -all  Convert the whole project\n";
+
+const dumpHelp =
+  "Usage: rescript dump <options> [target]\n" +
+  "`rescript dump` dumps the information for the target\n";
+
+// Shows build help with --help arg
+let out = child_process.spawnSync(`../../../rescript`, ["build", "--help"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stdout, buildHelp);
+
+// Shows build help with -h arg
+out = child_process.spawnSync(`../../../rescript`, ["build", "-h"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stdout, buildHelp);
+
+// Exits with build help with unknown arg
+out = child_process.spawnSync(`../../../rescript`, ["build", "-wtf"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stderr, "Error: unknown option: '-wtf'.\n" + buildHelp + "\n");
+
+// Shows cli help with --help arg
+out = child_process.spawnSync(`../../../rescript`, ["--help"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stdout, cliHelp);
+
+// Shows cli help with -h arg
+out = child_process.spawnSync(`../../../rescript`, ["-h"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stdout, cliHelp);
+
+// Shows cli help with help command
+out = child_process.spawnSync(`../../../rescript`, ["help"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+assert.equal(out.stdout, cliHelp);
+
+// Shows cli help with unknown command
+out = child_process.spawnSync(`../../../rescript`, ["built"], {
+  encoding: "utf8",
+  cwd: __dirname,
+});
+// Should write to stderr instead ???
+assert.equal(out.stdout, cliHelp);
```

Comment:
Should we write to `stderr` instead?

Reply:
I would say yes, to be consistent for all error cases.
I saw that an exit code is set, too, in this case, it would be great if the tests could check the exit code, too.